search/src/dockit-search.ts

```diff
@@ -0,0 +1,260 @@
+import { LitElement, css, html } from 'lit';
+import { unsafeHTML } from 'lit/directives/unsafe-html.js';
+
+const beforeHitMatch = 20;
+const afterHitMatch = beforeHitMatch;
+const markChar = '\0';
+
+export class SearchBox extends LitElement {
+  search = null;
+  hits = null;
+
+  constructor() {
+    super();
+    addEventListener('keydown', (e) => {
+      if ((e.ctrlKey || e.metaKey) && e.key === 'k') {
+        e.preventDefault();
+        this.updateHits();
+      } else if (e.key === 'Escape') this.hideHits();
+    });
```

Comment:
IIRC putting event listener inside the constructor avoid the the connected/disconnected callback because the lambda will be garbage collected once the Object is derefencered. Not 100% sure

Reply:
you are adding an event listener on the window, it's never gonna be garbage collected automatically

Reply:
true